pkg/datasources/shares.go

```diff
@@ -63,36 +63,35 @@ func Shares() *schema.Resource {
 func ReadShares(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
 	d.SetId("shares_read")
-	sharePattern := d.Get(pattern).(string)
-
-	listShares, err := snowflake.ListShares(db, sharePattern)
-	if errors.Is(err, sql.ErrNoRows) {
-		log.Printf("[DEBUG] no shares found in account (%s)", d.Id())
-		d.SetId("")
-		return nil
-	} else if err != nil {
-		log.Println("[DEBUG] failed to list shares")
-		d.SetId("")
-		return nil
+	pattern := d.Get(pattern).(string)
```

Comment:
I don't understand this, how can `pattern` depend on itself?

Reply:
if pattern was set, or not empty string then we do `SHOW SHARES LIKE '<pattern>'` otherwise we do `SHOW SHARES`. theres no dependency

Reply:
confusing variable nane